src/utils/token_to_fiat_oracle.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from 'chai';
+import { stub } from 'sinon';
+import type { CoinGeckoPriceResponseData } from './coingecko_types';
+import type { FiatID, TokenID } from './fiat_oracle_types';
+import { TokenFiatPair } from './token_fiat_price';
+import { TokenToFiatOracle } from './token_to_fiat_oracle';
+
+const token: TokenID = 'ARWEAVE';
+const fiat: FiatID = 'USD';
+const testingPair = new TokenFiatPair(fiat, token);
+const testingCacheLifespan = 1000 * 2; // 2 sec
+const halfOfTheTestingLifespan = testingCacheLifespan / 2;
+
+const examplePriceValue = 15.05;
+const coinGeckoPriceResponseForPair: CoinGeckoPriceResponseData = {
+	[token]: {
+		[fiat]: examplePriceValue
+	}
+};
+const mockResponse = new Response(
+	new Blob([JSON.stringify(coinGeckoPriceResponseForPair)], { type: 'application/json' })
+);
+
+describe('The fiat oracle', () => {
+	const stubbedFetch = stub({ fetch });
+	let oracle: TokenToFiatOracle;
+
+	before(() => {
```

Comment:
We discovered that `before()` seems to behave more like `beforeAll()`. Use `beforeEach()` if your intention is to use a new oracle one each test

Reply:
It was intentional. I know `before()` might be confusing, I could use `beforeAll()` instead